backend/auth/auth.go

```diff
@@ -17,22 +17,42 @@ package auth
 import (
 	"context"
 	"net/http"
+	"peoplemath/models"
 	"strings"
 )
 
 type Auth interface {
-	Authenticate(ctx context.Context, token string) (userEmail string, err error)
-	Authorize(next http.HandlerFunc) http.HandlerFunc
+	Authenticate(next http.HandlerFunc) http.HandlerFunc
+	CanActOnTeam(user User, team models.Team, action string) bool
+	CanActOnTeamList(user User, generalPermissions models.GeneralPermissions, action string) bool
+}
+
+const (
+	ActionRead         = "read"
+	ActionWrite        = "write"
+	ActionReadTeamList = "readTeamList"
```

Comment:
These two actions seem redundant - can you just use `ActionRead` and `ActionWrite` in your `CanActOnTeamList` method?

Reply:
Yep, makes more sense.